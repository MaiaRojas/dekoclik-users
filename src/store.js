import { createStore, applyMiddleware, compose } from 'redux';
import { reactReduxFirebase } from 'react-redux-firebase';
import { reduxFirestore } from 'redux-firestore';
import firebase from 'firebase/app';
import 'firebase/auth';
import 'firebase/firestore';
import reducers from './reducers';
import apiMiddleware from './middleware';


firebase.initializeApp({
  apiKey: process.env.FIREBASE_API_KEY,
  authDomain: `${process.env.FIREBASE_PROJECT}.firebaseapp.com`,
  databaseURL: `https://${process.env.FIREBASE_PROJECT}.firebaseio.com`,
  projectId: process.env.FIREBASE_PROJECT,
  storageBucket: `${process.env.FIREBASE_PROJECT}.appspot.com`,
  messagingSenderId: process.env.FIREBASE_MESSAGING_SENDER_ID,
});


const firestore = firebase.firestore();
firestore.settings({ timestampsInSnapshots: true });


const middlewares = [apiMiddleware];


// Add redux Firebase to compose
const createStoreWithFirebase = compose(
  reactReduxFirebase(firebase, {
    userProfile: 'users',
    useFirestoreForProfile: true,
  }),
  reduxFirestore(firebase),
)(createStore);


// if (process.env.NODE_ENV !== 'production') {
  // eslint-disable-next-line global-require
  // const { logger } = require('redux-logger');
  // middlewares.push(logger);
// }


export default createStoreWithFirebase(reducers, applyMiddleware(...middlewares));
