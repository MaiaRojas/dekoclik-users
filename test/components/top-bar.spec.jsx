import React from 'react';
import { render } from 'enzyme';
import { Provider } from 'react-redux';
import TopBar from '../../src/components/top-bar';
import store from '../../src/store';


describe('<TopBar />', () => {

  it('should print warning in console.error when missing title', () => {
    const spy = jest.spyOn(console, 'error').mockImplementation(jest.fn());
    render(<Provider store={store}><TopBar /></Provider>);

    expect(spy.mock.calls.length > 0).toBe(true);
    expect(spy.mock.calls[0]).toMatchSnapshot();

    spy.mockReset();
    spy.mockRestore();
  });

  it('should render to a <header> containing an <h2> with the title', () => {
    const component = render((
      <Provider store={store}>
        <TopBar title="Blah blah blah" />
      </Provider>
    ));

    expect(component.get(0).tagName).toBe('header');
    expect(component.get(0).attribs).toMatchSnapshot();
    expect(component.find('h2').text()).toBe('Blah blah blah');
  });

  it('should have classes prop injected and title prop on instance', () => {
    const component = render((
      <Provider store={store}>
        <TopBar title="foo" />
      </Provider>
    ));
    expect(component).toHaveLength(1);
    const $h2 = component[0].children[0].children[1];
    expect(component[0].attribs.class).toMatch(/TopBar/);
    expect($h2.children[0].type).toBe('text');
    expect($h2.children[0].data).toBe('foo');
  });

});
