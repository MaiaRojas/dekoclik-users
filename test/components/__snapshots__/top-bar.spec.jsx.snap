// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopBar /> should print warning in console.error when missing title 1`] = `
Array [
  "Warning: Failed prop type: The prop \`title\` is marked as required in \`TopBar\`, but its value is \`undefined\`.
    in TopBar
    in WithStyles(TopBar)
    in Connect(WithStyles(TopBar))
    in Provider",
]
`;

exports[`<TopBar /> should render to a <header> containing an <h2> with the title 1`] = `
Object {
  "class": "MuiPaper-root-16 MuiPaper-elevation4-22 MuiAppBar-root-8 MuiAppBar-positionAbsolute-10 MuiAppBar-colorPrimary-14 TopBar-appBar-1",
}
`;
